### PR TITLE
Add usize support

### PR DESCRIPTION
**Pull request by FrostyHitoshura**
Note: only the "little/64" path was actually tested on my machine.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -575,6 +576,51 @@ mod tests {
         assert_eq!(bytes.as_slice_of::<u64>(), Ok(slice.as_ref()));
     }
 
+    #[test]
+    fn usize() {
+        let slice: [usize; 2] = [0, 1];
+        let bytes = slice.as_byte_slice();
+
+        if cfg!(target_endian = "big") {
+            if cfg!(target_pointer_width = "16") {
```
Comment (sdroege) [inline src/lib.rs]: There's 16 bit pointer support in Rust now? Nice :) DOS or for some microcontroller or ...?

**Comment by sdroege** [comment]
I'll do a release with this some time in the next days, unless it's urgent for you. Let me know :)

Thanks for the PR!

File: src/lib.rs (hunk 1 above)
Comment (FrostyHitoshura) [reply src/lib.rs]: I determined all the supported "usize" values reading https://doc.rust-lang.org/src/core/num/mod.rs.html#4566

I'm unsure which 16 bits platform are supported but at the moment, with "std" in the mix, I feel that the generated binary is too large to support a 16 bits platform properly. If I was to do DOS development, I would probably stick to C for the time being, even if I find Rust generally great!

**Comment by FrostyHitoshura** [comment]
Thanks for the very quick turnaround for the merge. My code which use your library is currently unpublished and not working too well so no rush to release a new version to accommodate my work but thanks for caring!

**Comment by sdroege** [comment]
Ok, great :) Let me know when/if it becomes urgent!

I'll also add `isize` in the meantime.

**Pull request merged**